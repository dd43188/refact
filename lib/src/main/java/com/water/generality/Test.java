package com.water.generality;

public class Test {


    private boolean printToPdf;
    private boolean printToDoc;



    public void someMethod() {

        if (printToPdf) {
            //implementation
        } else if (printToDoc) {
            //should implementation soon
        }

    }

}
