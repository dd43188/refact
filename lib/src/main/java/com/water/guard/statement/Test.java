package com.water.guard.statement;

public class Test {

    void func() {
        if (getConditonA()) {
            //do nothing or return
        } else {
            if (getConditonB()) {
                //do nothing or return
            } else {
                doConditionB();
            }
        }
    }

    private void doConditionB() {
    }

    private boolean getConditonB() {
        return true;
    }

    private boolean getConditonA() {
        return true;
    }
}
