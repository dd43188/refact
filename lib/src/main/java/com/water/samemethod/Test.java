package com.water.samemethod;

import java.awt.Rectangle;
import java.awt.geom.Rectangle2D;
import java.util.Stack;

public class Test {
    Rectangle rectangle;
    Rectangle2D rectangle2D;
    Stack stack;
}
