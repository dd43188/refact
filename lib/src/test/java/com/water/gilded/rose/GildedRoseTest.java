package com.water.gilded.rose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.Test;

public class GildedRoseTest {

    @Test
    public void updateQuality() {

        CombinationApprovals.verifyAllCombinations(
                this::doUpdate,
                new String[]{"foo", "Aged Brie", "Sulfuras, Hand of Ragnaros", "hjm", /**"Backstage passes to a TAFKAL80ETC concert"**/},
                new Integer[]{-1, 0, 5, 6, 10, 11},
                new Integer[]{0, 1, 49, 50});
    }

    private String doUpdate(String name, int sellIn, int quality) {
        Item[] items = new Item[]{new Item(name, sellIn, quality)};
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        return items[0].toString();
    }
}