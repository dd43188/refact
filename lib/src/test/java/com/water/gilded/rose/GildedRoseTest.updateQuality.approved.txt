[foo, -1, 0] => foo, -2, 0 
[foo, -1, 1] => foo, -2, 0 
[foo, -1, 49] => foo, -2, 47 
[foo, -1, 50] => foo, -2, 48 
[foo, 0, 0] => foo, -1, 0 
[foo, 0, 1] => foo, -1, 0 
[foo, 0, 49] => foo, -1, 47 
[foo, 0, 50] => foo, -1, 48 
[foo, 5, 0] => foo, 4, 0 
[foo, 5, 1] => foo, 4, 0 
[foo, 5, 49] => foo, 4, 48 
[foo, 5, 50] => foo, 4, 49 
[foo, 6, 0] => foo, 5, 0 
[foo, 6, 1] => foo, 5, 0 
[foo, 6, 49] => foo, 5, 48 
[foo, 6, 50] => foo, 5, 49 
[foo, 10, 0] => foo, 9, 0 
[foo, 10, 1] => foo, 9, 0 
[foo, 10, 49] => foo, 9, 48 
[foo, 10, 50] => foo, 9, 49 
[foo, 11, 0] => foo, 10, 0 
[foo, 11, 1] => foo, 10, 0 
[foo, 11, 49] => foo, 10, 48 
[foo, 11, 50] => foo, 10, 49 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 1] => Aged Brie, -2, 3 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 5, 0] => Aged Brie, 4, 1 
[Aged Brie, 5, 1] => Aged Brie, 4, 2 
[Aged Brie, 5, 49] => Aged Brie, 4, 50 
[Aged Brie, 5, 50] => Aged Brie, 4, 50 
[Aged Brie, 6, 0] => Aged Brie, 5, 1 
[Aged Brie, 6, 1] => Aged Brie, 5, 2 
[Aged Brie, 6, 49] => Aged Brie, 5, 50 
[Aged Brie, 6, 50] => Aged Brie, 5, 50 
[Aged Brie, 10, 0] => Aged Brie, 9, 1 
[Aged Brie, 10, 1] => Aged Brie, 9, 2 
[Aged Brie, 10, 49] => Aged Brie, 9, 50 
[Aged Brie, 10, 50] => Aged Brie, 9, 50 
[Aged Brie, 11, 0] => Aged Brie, 10, 1 
[Aged Brie, 11, 1] => Aged Brie, 10, 2 
[Aged Brie, 11, 49] => Aged Brie, 10, 50 
[Aged Brie, 11, 50] => Aged Brie, 10, 50 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 1] => Sulfuras, Hand of Ragnaros, -1, 1 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Sulfuras, Hand of Ragnaros, -1, 50] => Sulfuras, Hand of Ragnaros, -1, 50 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras, Hand of Ragnaros, 0, 1 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 5, 0] => Sulfuras, Hand of Ragnaros, 5, 0 
[Sulfuras, Hand of Ragnaros, 5, 1] => Sulfuras, Hand of Ragnaros, 5, 1 
[Sulfuras, Hand of Ragnaros, 5, 49] => Sulfuras, Hand of Ragnaros, 5, 49 
[Sulfuras, Hand of Ragnaros, 5, 50] => Sulfuras, Hand of Ragnaros, 5, 50 
[Sulfuras, Hand of Ragnaros, 6, 0] => Sulfuras, Hand of Ragnaros, 6, 0 
[Sulfuras, Hand of Ragnaros, 6, 1] => Sulfuras, Hand of Ragnaros, 6, 1 
[Sulfuras, Hand of Ragnaros, 6, 49] => Sulfuras, Hand of Ragnaros, 6, 49 
[Sulfuras, Hand of Ragnaros, 6, 50] => Sulfuras, Hand of Ragnaros, 6, 50 
[Sulfuras, Hand of Ragnaros, 10, 0] => Sulfuras, Hand of Ragnaros, 10, 0 
[Sulfuras, Hand of Ragnaros, 10, 1] => Sulfuras, Hand of Ragnaros, 10, 1 
[Sulfuras, Hand of Ragnaros, 10, 49] => Sulfuras, Hand of Ragnaros, 10, 49 
[Sulfuras, Hand of Ragnaros, 10, 50] => Sulfuras, Hand of Ragnaros, 10, 50 
[Sulfuras, Hand of Ragnaros, 11, 0] => Sulfuras, Hand of Ragnaros, 11, 0 
[Sulfuras, Hand of Ragnaros, 11, 1] => Sulfuras, Hand of Ragnaros, 11, 1 
[Sulfuras, Hand of Ragnaros, 11, 49] => Sulfuras, Hand of Ragnaros, 11, 49 
[Sulfuras, Hand of Ragnaros, 11, 50] => Sulfuras, Hand of Ragnaros, 11, 50 
[hjm, -1, 0] => hjm, -2, 0 
[hjm, -1, 1] => hjm, -2, 0 
[hjm, -1, 49] => hjm, -2, 47 
[hjm, -1, 50] => hjm, -2, 48 
[hjm, 0, 0] => hjm, -1, 0 
[hjm, 0, 1] => hjm, -1, 0 
[hjm, 0, 49] => hjm, -1, 47 
[hjm, 0, 50] => hjm, -1, 48 
[hjm, 5, 0] => hjm, 4, 0 
[hjm, 5, 1] => hjm, 4, 0 
[hjm, 5, 49] => hjm, 4, 48 
[hjm, 5, 50] => hjm, 4, 49 
[hjm, 6, 0] => hjm, 5, 0 
[hjm, 6, 1] => hjm, 5, 0 
[hjm, 6, 49] => hjm, 5, 48 
[hjm, 6, 50] => hjm, 5, 49 
[hjm, 10, 0] => hjm, 9, 0 
[hjm, 10, 1] => hjm, 9, 0 
[hjm, 10, 49] => hjm, 9, 48 
[hjm, 10, 50] => hjm, 9, 49 
[hjm, 11, 0] => hjm, 10, 0 
[hjm, 11, 1] => hjm, 10, 0 
[hjm, 11, 49] => hjm, 10, 48 
[hjm, 11, 50] => hjm, 10, 49 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Backstage passes to a TAFKAL80ETC concert, 4, 3 
[Backstage passes to a TAFKAL80ETC concert, 5, 1] => Backstage passes to a TAFKAL80ETC concert, 4, 4 
[Backstage passes to a TAFKAL80ETC concert, 5, 49] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 50] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 0] => Backstage passes to a TAFKAL80ETC concert, 5, 2 
[Backstage passes to a TAFKAL80ETC concert, 6, 1] => Backstage passes to a TAFKAL80ETC concert, 5, 3 
[Backstage passes to a TAFKAL80ETC concert, 6, 49] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 50] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Backstage passes to a TAFKAL80ETC concert, 9, 2 
[Backstage passes to a TAFKAL80ETC concert, 10, 1] => Backstage passes to a TAFKAL80ETC concert, 9, 3 
[Backstage passes to a TAFKAL80ETC concert, 10, 49] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Backstage passes to a TAFKAL80ETC concert, 10, 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] => Backstage passes to a TAFKAL80ETC concert, 10, 2 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
